watcher/watcher.go

```diff
@@ -368,8 +368,6 @@ func (watcher Watcher) watchLogShiftOuts(parent context.Context) {
 		amount := burn.Amount
 		to := burn.ToBytes
 
-		watcher.logger.Infof("[watcher] detected burn for %v (to=%v, amount=%v, nonce=%v)", watcher.selector.String(), string(to), amount, nonce)
```

Comment:
I slightly prefer to leave the detection message here, as if we fail to construct the params, we won't know which selector failed (we could alternately add the selector to the error handler for burnToParams; but conceptually we have still "detected" the burn successfully, so logging it here still makes sense)

Reply:
Fair point, I moved it to `burnToParams` because I was under the assumption that solana expects raw addresses only, which is why at this point `to=%v` outputs all kinds of weird characters. However, now that I think about it, it doesn't matter what format is passed in to solana (although better to use a raw address, since it will be shorter than an encoded one). Maybe we can break this up into two parts, here it logs that a burn was detected, and in `burnToParams` it logs the relevant info after the address has been encoded to a readable format

Reply:
Yes, good point with the garbled `to` address; and happy with logging the details in `burnToParams` - although we still want to capture at least the selector and the nonce in the detection message (because we can manually recover the details from the burn program with those details)